### ISSUE TITLE
Decoding AlgorithmIdentifier.parameters was skipped

### ISSUE THREAD

**Issue by shigeki** [indutny/asn1.js#40]
Decoding a cert with asn1.js-rfc3280 to extract `subjectPublicKeyInfo` field shows that `AlgorithmIdentifier.parameters` was missed, where 0x05,0x00(NULL) was set in the case of RSA.  The field was parsed in the next `subjectPublicKey` field in the `unused` property.
```js
{ algorithm: { algorithm: [ 1, 2, 840, 113549, 1, 1, 1 ] },
  subjectPublicKey:
   { unused: 0,
     data: <Buffer 30 82 01 0a 02 82 01 01 00 b7 89 4e 02 f9 ba 01 e0 78 89 d6 70 fd 36 18 d6 02 2e fc 96 c9 d9 de ae 2e 80 0a a1 9f 4b 17 20 c3 71 b9 99 6b 2e fc 12 fa ... > } }
````
It seems that the decoder cannot parse and obtain the field with defined with `optional().any()`.

This causes the problem when we want to extract the der data of `subjectPublicKey` from `tbsCertificate` by simply decoding and encoiding of `cert.tbsCertificate.subjectPublicKeyInfo` with
```js
var spki_der = SubjectPublicKeyInfo.encode(cert.tbsCertificate.subjectPublicKeyInfo, 'der');
````
so that NULL(0x05,0x00) was missed.
But we have a workaround to set the value of parameters explicitly.
```js
var spki = cert.tbsCertificate.subjectPublicKeyInfo;
var new_spki = {algorithm : {algorithm: spki.algorithm.algorithm, parameters: new Buffer('0500', 'hex')},
                           subjectPublicKey: {data: spki.subjectPublicKey.data}};
var spki_der = SubjectPublicKeyInfo.encode(new_spki, 'der');
````
If there is a more simple way to extract the der data, please let me know it.

**Comment by indutny** [comment]
@shigeki may I ask you to gist an HEX-encoding of certificate?

**Comment by shigeki** [comment]
@indutny Sure. We can confirm this in the test of rfc/3280/test with a RSA signed cert, that is Google G2, from

```diff
diff --git a/rfc/3280/test/basic-test.js b/rfc/3280/test/basic-test.js
index 128ddbb..e75b5a6 100644
--- a/rfc/3280/test/basic-test.js
+++ b/rfc/3280/test/basic-test.js
@@ -45,6 +45,7 @@ describe('asn1.js RFC3280', function() {
     var res = rfc3280.Certificate.decode(data, 'der');

     var tbs = res.tbsCertificate;
+    console.log(tbs.subjectPublicKeyInfo);
     assert.equal(tbs.version, 'v3');
     assert.deepEqual(tbs.serialNumber,
                      new asn1.bignum('462e4256bb1194dc', 16));
````
```sh
$ mocha rfc/3280/test/basic-test.js


  asn1.js RFC3280
{ algorithm: { algorithm: [ 1, 2, 840, 113549, 1, 1, 1 ] },
  subjectPublicKey:
   { unused: 0,
     data: <Buffer 30 82 01 0a 02 82 01 01 00 b7 89 4e 02 f9 ba 01 e0 78 89 d6 70 fd 36 18 d6 02 2e fc 96 c9 d9 de ae 2e 80 0a a1 9f 4b 17 20 c3 71 b9 99 6b 2e fc 12 fa ... > } }
    ? should decode Certificate


  1 passing (24ms)
````
According to RFC3279, the parameter of RSA is NULL(0x05,0x00) and it is in the der hex above.

For a DSA cert, I got an intermidate cert  from symantec site of in https://gist.github.com/shigeki/0a360e92a1652538aa66 . After converting it into der, the test shows no P, Q and G as below.
```js
var fs = require('fs');
var rfc3280 = require('asn1.js-rfc3280');

var Certificate = rfc3280.Certificate;

var der = fs.readFileSync('./dsa_cert.der');

var cert = Certificate.decode(der, 'der');
console.log(cert.tbsCertificate.subjectPublicKeyInfo);
````
```sh
$ ~/github/io.js/iojs test.js
{ algorithm: { algorithm: [ 1, 2, 840, 10040, 4, 1 ] },
  subjectPublicKey:
   { unused: 0,
     data: <Buffer 02 82 01 00 2f 1a e5 8f 76 cc 70 cf c0 2d a6 40 8e 4e d2 07 f6 b2 ce 04 89 63 ea 29 92 6d 39 42 0a e0 7d 8a 38 c3 42 91 e5 a6 6a b6 54 da 40 1f f7 e5 ... > } }
````

I looked around the code and found that in the node of `AlgorithmIdentifier.parameters`,  both of the `state.optional` and `state.any` was true but `state.tag` was null so that `present` was false that was returned from `_peekTag ()`  in lib/asn1/base/node.js. I hope it helps.

**Comment by indutny** [comment]
Yay, I see now! Thank you.

Fixed in 8957977.

**Comment by shigeki** [comment]
@indutny Thanks. I found an another issue relating to this fix. I will submit a new PR.
